library/src/main/java/com/novoda/downloadmanager/UnlinkedDataRemover.java

```diff
@@ -0,0 +1,25 @@
+package com.novoda.downloadmanager;
+
+import java.util.List;
+
+class UnlinkedDataRemover {
+
+    private final RoomAppDatabase database;
```

Comment:
This **must** be database agnostic, the rest of the code only talks to the interface `DownloadsPersistence`

Reply:
`DownloadsPersistence` is not database agnostic using this logic. If one place needs to do it then all should be expected to do it by using an interface. That was kind of my point.

Reply:
Ah! I didn't realise that `DownloadsPersistence` was the interface I was looking for 😂